src/libraries/System.Formats.Tar/tests/TarTestsBase.cs

```diff
@@ -108,6 +108,21 @@ protected static MemoryStream GetTarMemoryStream(CompressionMethod compressionMe
             return ms;
         }
 
+        protected static string GetStrangeTarFilePath(string testCaseName) =>
+            Path.Join(Directory.GetCurrentDirectory(), "strange", testCaseName + ".tar");
+
+        protected static MemoryStream GetStrangeTarMemoryStream(string testCaseName)
+        {
+            string path = GetStrangeTarFilePath(testCaseName);
+            MemoryStream ms = new();
```

Comment:
Can this be refactored with the code above it to reduce duplication?

Reply:
Sure.